src/test/java/com/powsybl/caseserver/datasource/CaseDataSourceControllerWithIidmTest.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.caseserver.datasource;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.powsybl.caseserver.ContextConfigurationWithTestChannel;
+import com.powsybl.caseserver.elasticsearch.DisableElasticsearch;
+import com.powsybl.caseserver.service.CaseService;
+import com.powsybl.caseserver.service.MinioContainerConfig;
+import com.powsybl.caseserver.service.S3CaseService;
+import com.powsybl.commons.datasource.DataSource;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.cloud.stream.function.StreamBridge;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.nio.file.Paths;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * @author Etienne Lesot <etienne.lesot at rte-france.com>
+ */
+@DisableElasticsearch
+@AutoConfigureMockMvc
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
+@TestPropertySource(properties = {"storage.type=S3"})
+@ContextConfigurationWithTestChannel
+class CaseDataSourceControllerWithIidmTest implements MinioContainerConfig {
```

Comment:
You need all those tests ? Isn't it redundant with FsCaseDataSourceControllerTest or S3CaseDataSourceControllerTest ?
If those tests are needed, we should have it for both implementation (FS and S3). You should implement them in AbstractCaseDataSourceControllerTest